src/coreclr/System.Private.CoreLib/src/System/Object.CoreCLR.cs

```diff
@@ -1,16 +1,39 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace System
 {
     public partial class Object
     {
         // Returns a Type object which represent this object instance.
         [Intrinsic]
+        public unsafe Type GetType()
+        {
+            // Throws NullReferenceException as expected
+            MethodTable* pMT = RuntimeHelpers.GetMethodTable(this);
+
+            Type? type = GetTypeIfExists(pMT);
```

Comment:
Would reusing the same variable pessimize JIT? The value may be seen as address exposed.

Reply:
This method is potentially sharable with GetTypeFromHandleUnsafe, but the unnecessary check for `TypeDesc` may hurt performance.